gremlin-go/driver/graph.go

```diff
@@ -120,3 +129,54 @@ func (p *Path) GetPathObject(key string) (interface{}, error) {
 		return nil, fmt.Errorf("path does not contain a label of '%s'", key)
 	}
 }
+
+// Set describes the necessary methods that need to be implemented to make a user-defined SimpleSet implementation.
+// SimpleSet is the default implementation that should be used but Set can be used for a custom implementation if needed.
+type Set interface {
+	SetToSlice() ([]interface{}, error)
```

Comment:
Nit: I still prefer this just to be `ToSlice` but am not going to block on it.

This also needs documentation.